components/social/org.wso2.carbon.social.sql/src/main/java/org/wso2/carbon/social/sql/SQLActivityBrowser.java

```diff
@@ -637,29 +649,70 @@ public JsonObject getPopularAssets(String type, String tenantId, int limit,
 
 	}
 
-	/**
-	 * Check whether the given user has already reviewed on the given targetId (AssetType + UUID)
-	 * @param targetId AssetType + UUID delimited by colon
-	 * @param userId Username with tenant domain i:e admin@carbon.super
-	 * @return Boolean whether user has already reviewed or not
-	 */
-    public boolean isReviewed(String targetId, String userId) throws SocialActivityException {
-        boolean reviewed = false;
+    /**
+     * Check whether the given user has already published the given activity on targetId (AssetType + UUID)
+     *
+     * @param activityString Stringify activity JSON, An activity can be either
+     *                       comment or like/dislike unlike/un-dislike
+     * @param targetId       AssetType + UUID delimited by colon
+     * @param userId         Username with tenant domain i:e admin@carbon.super
+     * @return Boolean whether user has already reviewed or not
+     */
+    public boolean isPublished(String activityString, String targetId, String userId) throws SocialActivityException {
+        boolean published;
+        JsonObject activityJSON;
+        int likeValue = -1;
+        try {
+            activityJSON = (JsonObject) parser.parse(activityString);
+        } catch (JsonSyntaxException e) {
+            String message = "Malformed JSON element found: " + e.getMessage();
+            log.error(message, e);
+            throw new SocialActivityException(message, e);
+        } catch (NumberFormatException e) {
+            String message = "Invalid review ID : " + e.getMessage();
+            throw new SocialActivityException(message, e);
+        }
+        SQLActivity activity = new SQLActivity(activityJSON);
+        String verb = activity.getVerb();
+        if (Constants.POST_VERB.equals(verb)) {
+            published = isCommented(targetId, userId);
+        } else {
+            if (Constants.VERB.valueOf(verb) == Constants.VERB.like) {
+                likeValue = 1;
+            } else if (Constants.VERB.valueOf(verb) == Constants.VERB.dislike) {
+                likeValue = 0;
+            }
```

Comment:
What I have done previously was, declared and assigned the  **likeValue**  to -1, so -1 would work as the default value to  **likeValue** . To improve the readability, Then I have moved the -1 assignment to **else** case where it assigns the -1 if no above cases satisfied to match.With that change  **likeValue**  doesn't  get assigned -1 as its default value.